python/core/auto_generated/vector/qgsvectordataprovider.sip.in

```diff
@@ -128,6 +128,19 @@ Returns ``True`` if the layer does not contain any feature.
 .. versionadded:: 3.4
 %End
 
+    virtual bool isQuery() const;
```

Comment:
I wonder if it'd be a bit more flexible to instead have a "layerType" method which returns an enum instead (initially with Table/Query values).

Reply:
Done